platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileMetadataJdb.java

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+
+package com.swirlds.merkledb.files;
+
+import static com.swirlds.merkledb.files.DataFileCommon.FOOTER_SIZE;
+import static com.swirlds.merkledb.files.DataFileCompactor.INITIAL_COMPACTION_LEVEL;
+
+import com.swirlds.base.utility.ToStringBuilder;
+import com.swirlds.merkledb.utilities.MerkleDbFileUtils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * DataFile's metadata that is stored in the data file's footer
+ */
+@SuppressWarnings("unused")
+// Future work: remove this class, once all DB files are converted to protobuf
+// https://github.com/hashgraph/hedera-services/issues/8344
+public final class DataFileMetadataJdb extends DataFileMetadata {
+
+    /**
+     * Create a new DataFileMetadata with complete set of data
+     *
+     * @param itemsCount           The number of data items the file contains
+     * @param index                The file index, in a data file collection
+     * @param creationDate         The creation data of this file, this is critical as it is used when merging two files
+     *                             to know which files data is newer.
+     * @param serializationVersion Serialization version for data stored in the file
+     */
+    public DataFileMetadataJdb(
+            final long itemsCount,
+            final int index,
+            final Instant creationDate,
+            final long serializationVersion,
+            final int compactionLevel) {
+        super(itemsCount, index, creationDate, serializationVersion, compactionLevel);
+    }
+
+    /**
+     * Create a DataFileMetadata loading it from a existing file
+     *
+     * @param file The file to read metadata from
+     * @throws IOException If there was a problem reading metadata footer from the file
+     */
+    public DataFileMetadataJdb(Path file) throws IOException {
+        super(0, 0, null, 0, INITIAL_COMPACTION_LEVEL);
+        try (final SeekableByteChannel channel = Files.newByteChannel(file, StandardOpenOption.READ)) {
+            // read footer from end of file
+            final ByteBuffer buf = ByteBuffer.allocate(FOOTER_SIZE);
+            channel.position(channel.size() - FOOTER_SIZE);
+            MerkleDbFileUtils.completelyRead(channel, buf);
+            buf.rewind();
+            // parse content
+            buf.getInt(); // backwards compatibility: file format version
+            buf.getInt(); // backwards compatibility: data item value size
+            this.itemsCount = buf.getLong();
+            this.index = buf.getInt();
+            this.creationDate = Instant.ofEpochSecond(buf.getLong(), buf.getInt());
+            this.compactionLevel = buf.get(); // backwards compatibility: used to be a byte for isMergeFile
```

Comment:
This field hasn't been used as 'isMergeFile' flag for many versions. So, this comment is more confusing than helpful. Usually we skip bytes for backward compatibilities or write some default values, but in this case we work with the field as usual.

Reply:
OK, let me remove the comment